script.js

```diff
@@ -1 +1,348 @@
+      }
+      if (i === 0) {
+        square.classList.add("top");
+      }
+      if (i === board.length - 1) {
+        square.classList.add("bottom");
+      }
+      // set the text of the square according to the array
+      square.innerText = board[i][j];
+      rowElement.appendChild(square);
+      // set the click all over again
+      // eslint-disable-next-line
+      square.addEventListener("click", () => {
+        squareClick(i, j);
+      });
+    }
+    // add a single row to the board
+    boardContainer.appendChild(rowElement);
+  }
+};
+//check empty cells
+const emptyCellCheck = () => {
+  emptyCells = [];
+  for (let i = 0; i < board.length; i++) {
+  for (let j = 0; j <board.length; j++) {
+    if (board[i][j] === "") {
+        emptyCells.push([i,j]);
+    }
+}}
+return emptyCells;
+}
+
+
+// find available corner Moves
+const emptyCornerCheck = () => {
+  cornerMove = [];
+  if (board[0][0] ==="") {
+    cornerMove.push([0,0]);
+  }
+  if (board[0][board.length-1] ==="") {
+    cornerMove.push([0,board.length-1]);
+  }
+    if (board[board.length-1][0] ==="") {
+    cornerMove.push([board.length-1,0]);
+  }
+  if (board[board.length-1][board.length-1] ==="") {
+    cornerMove.push([board.length-1,board.length-1]);
+  }
+  return cornerMove;
+}
+
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+const getRandomIndex = (max) => {
+  return Math.floor(Math.random() * max);
+};
+
+/*#############
+GAMEPLAY LOGIC
+##############*/
+
+// switch the global values from one player to the next
+const togglePlayer = (isComp) => {
+  emptyCellCheck();
+  emptyCornerCheck();
+  gameInfo.innerText = "";
+  if (checkWin()===true) {
+      gameInfo.innerText = `${currentPlayer} is winner`;
+      lockBoard = true;
+    }
+  if (gameTie() === false) {
+    if (currentPlayer === huPlayer) {
+    currentPlayer = aiPlayer;
+    isComp ? aiPlay() : null;
+  } else {
+    currentPlayer = huPlayer;
+  }
+  }
+  else {
+    gameInfo.innerText = "It's a tie";
+    lockBoard = true;
+  } 
+};
+
+// computer choose randomly
+const computerPlay = (anArray) => {
+  let randomIndex = getRandomIndex(anArray.length);
+  setTimeout (function () {
+    squareClick(anArray[randomIndex][0],anArray[randomIndex][1]);
+  },500);
+}
+
+const computerPick = () => {
+  //computer pick the coordinate that will either win the game or block X
+  if (winMove.length > 0) {
+    //console.log("win move is played");
+    computerPlay(winMove);
+  }
+  if (blockMove.length > 0 && winMove.length ===0) {
+    //console.log("block move is played");
+    computerPlay(blockMove);
+  }
+  if (cornerMove.length > 0 && blockMove.length ===0 && winMove.length ===0) {
+    //console.log("corner move is played");
+    computerPlay(cornerMove);
+  }
+  if (cornerMove.length === 0 && blockMove.length ===0 && winMove.length ===0) {
+    //console.log("empty cell is played");
+    computerPlay(emptyCells);}
+}
+
+const simulateClick = (player, moveArray) => {
```

Comment:
You can improve the developer-friendliness by giving `player` some context. Right now, for a new developer reading this code, they may wonder if `player` is referring to the player's move, player's choice, or the actual player's symbol (X / O)...etc